src/geometry/euclidean/oriented_angle.lean

```diff
@@ -587,332 +670,241 @@ end
-    norm_num at hd }
+  haveI : nontrivial V :=
+    finite_dimensional.nontrivial_of_finrank_eq_succ (fact.out (finrank ℝ V = 2)),
+  obtain ⟨x, hx⟩ : ∃ x, x ≠ (0:V) := exists_ne (0:V),
+  use o.oangle x (f x),
+  apply linear_isometry_equiv.to_linear_equiv_injective,
+  apply linear_equiv.to_linear_map_injective,
+  apply (o.basis_right_angle_rotation x hx).ext,
+  intros i,
+  symmetry,
+  fin_cases i,
+  { simp },
+  have : o.oangle (J x) (f (J x)) = o.oangle x (f x),
+  { simp only [oangle, o.linear_isometry_equiv_comp_right_angle_rotation f hd,
+      o.kahler_comp_right_angle_rotation] },
+  simp [← this],
 end
 
-/-- Any linear isometric equivalence in `V` with negative determinant is `conj_lie` composed
-with `rotation`. -/
-lemma exists_linear_isometry_equiv_eq_of_det_neg {f : V ≃ₗᵢ[ℝ] V}
-  (hd : (f.to_linear_equiv : V →ₗ[ℝ] V).det < 0) :
-  ∃ θ : real.angle, f = hb.conj_lie.trans (hb.rotation θ) :=
-begin
-  rcases hb.exists_linear_isometry_equiv_eq f with ⟨θ, (hf|hf)⟩,
-  { simp [hf, ←linear_equiv.coe_det] at hd,
-    norm_num at hd },
-  { exact ⟨θ, hf⟩ }
-end
+/-- The angle between two vectors, with respect to an orientation given by `orientation.map`
+with a linear isometric equivalence, equals the angle between those two vectors, transformed by
+the inverse of that equivalence, with respect to the original orientation. -/
+@[simp] lemma oangle_map (x y : V') (f : V ≃ₗᵢ[ℝ] V') :
+  (orientation.map (fin 2) f.to_linear_equiv o).oangle x y = o.oangle (f.symm x) (f.symm y) :=
+by simp [oangle, o.kahler_map]
 
-/-- Two bases with the same orientation are related by a `rotation`. -/
-lemma exists_linear_isometry_equiv_map_eq_of_orientation_eq {b₂ : basis (fin 2) ℝ V}
-  (hb₂ : orthonormal ℝ b₂) (ho : b.orientation = b₂.orientation) :
-  ∃ θ : real.angle, b₂ = b.map (hb.rotation θ).to_linear_equiv :=
-begin
-  have h : b₂ = b.map (hb.equiv hb₂ (equiv.refl _)).to_linear_equiv,
-  { rw hb.map_equiv, simp },
-  rw [eq_comm, h, b.orientation_comp_linear_equiv_eq_iff_det_pos] at ho,
-  cases hb.exists_linear_isometry_equiv_eq_of_det_pos ho with θ hθ,
-  rw hθ at h,
-  exact ⟨θ, h⟩
-end
+@[simp] protected lemma _root_.complex.oangle (w z : ℂ) :
+  complex.orientation.oangle w z = complex.arg (conj w * z) :=
+by simp [oangle]
 
-/-- Two bases with opposite orientations are related by `conj_lie` composed with a `rotation`. -/
-lemma exists_linear_isometry_equiv_map_eq_of_orientation_eq_neg {b₂ : basis (fin 2) ℝ V}
-  (hb₂ : orthonormal ℝ b₂) (ho : b.orientation = -b₂.orientation) :
-  ∃ θ : real.angle, b₂ = b.map (hb.conj_lie.trans (hb.rotation θ)).to_linear_equiv :=
+/-- The oriented angle on an oriented real inner product space of dimension 2 can be evaluated in
+terms of a complex-number representation of the space. -/
+lemma oangle_map_complex (f : V ≃ₗᵢ[ℝ] ℂ)
+  (hf : (orientation.map (fin 2) f.to_linear_equiv o) = complex.orientation) (x y : V) :
+  o.oangle x y = complex.arg (conj (f x) * f y) :=
```

Comment:
I think some lemmas are missing that could be used as `hf` to make use of this and similar lemmas. That is, I think there should be a lemma (not in this file) about mapping the orientation given by an orthonormal basis with `(complex.isometry_of_orthonormal b).symm` giving `complex.orientation` (and the reverse direction lemma, that mapping `complex.orientation` with `complex.isometry_of_orthonormal b` gives the orientation given by `b`, is surely also an appropriate lemma though not directly relevant here). And variants where the basis is `o.fin_orthonormal_basis` would also be appropriate.

Reply:
Agreed.  I had deferred this out of laziness, but it should be easy, and you're right that it's needed for usability.  I'll add it.

Another TODO (but not for this PR) is to define a predicate `orientation_preserving`, so that we can say the shorter and more readable
```lean
orientation_preserving f.to_linear_equiv o complex.orientation
```
instead of
```lean
(orientation.map (fin 2) f.to_linear_equiv o) = complex.orientation
```